### PR TITLE
Indicate eviction works by returning a boolean result)

### PR DESCRIPTION
**Pull request by ahmetmircik**
This change is needed for https://github.com/hazelcast/hazelcast-enterprise/pull/3497

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/NearCacheRecordStore.java [hunk 1]
```diff
@@ -110,7 +110,7 @@
      *                            size check, otherwise set {@code false} to evict
      *                            after a max size check.
      */
-    void doEviction(boolean withoutMaxSizeCheck);
+    boolean doEviction(boolean withoutMaxSizeCheck);
```
Comment (vojtechtoman) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/NearCacheRecordStore.java]: minor: internal API but adding a `@return` would be nice

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java [hunk 2]
```diff
@@ -321,15 +321,16 @@ public int size() {
...
         checkAvailable();
 
         if (evictionDisabled) {
-            return;
+            return false;
         }
 
         EvictionChecker evictionChecker = withoutMaxSizeCheck ? null : this.evictionChecker;
         evictionStrategy.evict(records, evictionPolicyEvaluator, evictionChecker, this);
+        return true;
```
Comment (vojtechtoman) [inline hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java]: Just to double-check - the idea is to return true if eviction was performed, but not if eviction was performed **and** an entry was evicted (that is, return the result of `evictionStrategy.evict()`), correct?

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/NearCacheRecordStore.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/internal/nearcache/NearCacheRecordStore.java]: fixed

File: hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java (hunk 2 above)
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java]: your understanding is true.

**Pull request merged**